src/components/ha-form/types.ts

```diff
@@ -9,14 +10,24 @@ export type HaFormSchema =
   | HaFormBooleanSchema
   | HaFormSelectSchema
   | HaFormMultiSelectSchema
-  | HaFormTimeSchema;
+  | HaFormTimeSchema
+  | HaFormSelector;
 
 export interface HaFormBaseSchema {
   name: string;
+  // This value is applied if no data is submitted for this field
   default?: HaFormData;
   required?: boolean;
   optional?: boolean;
```

Comment:
We should decided which one to use by default, maybe drop 1 from the type

Reply:
I would say `required`

Reply:
Sure, but let's do that in a future PR. We will need to check what is used where. 

For the backend, both required and optional fields can have a default, which is filled in when key is omitted. So not so required or optional…

Reply:
I agree with keeping `required`. Let's update it in another PR. It means rewriting all checks for `optional` to be `!required`.

Reply:
https://github.com/home-assistant/frontend/pull/11538